api/utils/grpc/stream/stream_test.go

```diff
@@ -122,6 +122,30 @@ func TestReadWriter_WriteChunk(t *testing.T) {
 	wg.Wait()
 }
 
+func TestReadWriter_WriteDisabledChunk(t *testing.T) {
+	streamConn, local := newStreamPipe(t, WithDisabledChunking())
+	wg := &sync.WaitGroup{}
+	wg.Add(2)
+
+	data := make([]byte, MaxChunkSize+1)
+	go func() {
+		defer wg.Done()
+		n, err := streamConn.Write(data)
+		assert.NoError(t, err)
+		assert.Len(t, data, n)
+	}()
+	go func() {
+		defer wg.Done()
+		b := make([]byte, 2*MaxChunkSize)
+		n, err := local.Read(b)
+		assert.NoError(t, err)
+		assert.Len(t, data, n)
+		assert.Equal(t, data[:n], b[:n])
```

Comment:
Isn't this racy? Writing to data in one goroutine and reading in another.

Reply:
The goroutines only read from `data` and do not modify it, so there shouldn't be a race condition?
`go test -race` also didn't find anything.